ocsgw/src/main/java/com/telenordigital/ocsgw/data/grpc/GrpcDataSource.java

```diff
@@ -183,10 +181,22 @@ private CreditControlAnswer createCreditControlAnswer(CreditControlAnswerInfo re
         final LinkedList<MultipleServiceCreditControl> multipleServiceCreditControls = new LinkedList<>();
         for (com.telenordigital.prime.ocs.MultipleServiceCreditControl mscc : response.getMsccList() ) {
             multipleServiceCreditControls.add(convertMSCC(mscc));
+            updateBlockedList(mscc, response.getMsisdn());
         }
         return new CreditControlAnswer(multipleServiceCreditControls);
     }
 
+    private void updateBlockedList(com.telenordigital.prime.ocs.MultipleServiceCreditControl msccGRPC, String msisdn) {
+        // This suffers from the fact that one Credit-Control-Request can have multiple MSCC
+        if (msccGRPC != null && msisdn != null) {
+            if (msccGRPC.getGranted().getTotalOctets() == 0) {
```

Comment:
Not only zero. Even when the granted is greater than zero, it MAY be blocked.
A better condition would be if "granted < requested".
This would imply that this might be the last bucket.